src/Microsoft.ML.AutoML/AutoMLExperiment/AutoMLExperiment.cs

```diff
@@ -276,21 +294,25 @@ void handler(object o, EventArgs e)
                         {
                             aggregateTrainingStopManager.OnStopTraining += handler;
 
-                            performanceMonitor.MemoryUsageInMegaByte += (o, m) =>
+                            performanceMonitor.PerformanceMetricsUpdated += (o, metrics) =>
                             {
-                                if (_settings.MaximumMemoryUsageInMegaByte is double d && m > d && !trialCancellationTokenSource.IsCancellationRequested)
+                                trialSettings.PerformanceMetrics = metrics;
```

Comment:
After moving `ReportTrialResourceUsage` from `IMonitor` to `IPerformanceMonitor`, we can call `performanceMonitor.ReportTrialResourceUsage` in this handler